src/pages/NewGroupPage.js

```diff
@@ -183,62 +179,65 @@ class NewGroupPage extends Component {
             maxParticipantsReached,
         );
         return (
-            <>
-                <HeaderGap />
-                <HeaderWithCloseButton
-                    title="New Group"
-                    onCloseButtonPress={redirectToLastReport}
-                />
-                <View style={[styles.flex1, styles.w100]}>
-                    <OptionsSelector
-                        canSelectMultipleOptions
-                        sections={sections}
-                        selectedOptions={this.state.selectedOptions}
-                        value={this.state.searchValue}
-                        onSelectRow={this.toggleOption}
-                        onChangeText={(searchValue = '') => {
-                            const {
-                                recentReports,
-                                personalDetails,
-                                userToInvite,
-                            } = getNewGroupOptions(
-                                this.props.reports,
-                                this.props.personalDetails,
-                                searchValue,
-                                [],
-                            );
-                            this.setState({
-                                searchValue,
-                                userToInvite,
-                                recentReports,
-                                personalDetails,
-                            });
-                        }}
-                        headerMessage={headerMessage}
-                        disableArrowKeysActions
-                        hideAdditionalOptionStates
-                        forceTextUnreadStyle
-                    />
-                    {this.state.selectedOptions?.length > 0 && (
-                        <View style={[styles.ph5, styles.pb5]}>
-                            <Pressable
-                                onPress={this.createGroup}
-                                style={({hovered}) => [
-                                    styles.button,
-                                    styles.buttonSuccess,
-                                    styles.w100,
-                                    hovered && styles.buttonSuccessHovered,
-                                ]}
-                            >
-                                <Text style={[styles.buttonText, styles.buttonSuccessText]}>
-                                    Create Group
-                                </Text>
-                            </Pressable>
+            <ScreenWrapper>
+                {() => (
```

Comment:
I was seeing all of these, and wondering if there is a way to clean these up a little. I might not have all the context behind this, but I looked through the docs for the safe area context: https://github.com/th3rdwave/react-native-safe-area-context

It appears to me that the only reason the `{() => ()}` pattern is used is because `ScreenWrapper` is using `SafeAreaInsetsContext.Consumer` to get access to the `insets`. The insets are only used in a single component (sidebar).

I'm curious if all the other views could be much more simple that don't need access to the insets. Maybe something like this:

```
// Normal screenwrapper defaults to not providing insets
<ScreenWrapper>
    <View />
</ScreenWrapper>
```

Then for the one that needs insets:
```
<ScreenWrapperWithInsets>
    {(insets) => (...)}
</ScreenWrapperWithInsets>
```